File: src/main/java/net/ndrei/teslacorelib/gui/PlayerInventoryBackground.java
package net.ndrei.teslacorelib.gui;

import net.ndrei.teslacorelib.containers.BasicTeslaContainer;

/**
 * Created by CF on 2016-12-22.
 */
public class PlayerInventoryBackground extends BasicContainerGuiPiece {
    public PlayerInventoryBackground(int left, int top, int width, int height) {
        super(left, top, width, height);
    }

    @Override
    public void drawBackgroundLayer(BasicTeslaGuiContainer container, int guiX, int guiY, float partialTicks, int mouseX, int mouseY) {
        BasicTeslaContainer slots = container.getTeslaContainer();
        if ((slots == null) || !slots.hasPlayerInventory()) {
            return;
        }

        container.bindDefaultTexture();

        int topOffset = 0;
        for(int i = 0; i < 3; i++) {
            container.drawTexturedModalRect(guiX + this.getLeft(), guiY + this.getTop() + topOffset, 7, 159,
                    Math.min(this.getWidth(), 162),
                    Math.min(this.getHeight() - topOffset, 18));
            topOffset += 18;
            if (topOffset > this.getHeight()) {
                break;
            }
        }
    }
}

File: src/main/java/net/ndrei/teslacorelib/gui/PlayerInventoryBackground.kt
package net.ndrei.teslacorelib.gui

/**
 * Created by CF on 2017-06-28.
 */
class PlayerInventoryBackground(left: Int, top: Int, width: Int, height: Int)
    : BasicContainerGuiPiece(left, top, width, height) {

    override fun drawBackgroundLayer(container: BasicTeslaGuiContainer<*>, guiX: Int, guiY: Int, partialTicks: Float, mouseX: Int, mouseY: Int) {
        val slots = container.teslaContainer
        if (slots == null || !slots.hasPlayerInventory()) {
            return
        }

        container.bindDefaultTexture()

        var topOffset = 0
        for (i in 0..2) {
            container.drawTexturedModalRect(guiX + this.left, guiY + this.top + topOffset, 7, 159,
                    Math.min(this.width, 162),
                    Math.min(this.height - topOffset, 18))
            topOffset += 18
            if (topOffset > this.height) {
                break
            }
        }
    }
}
